README.md

```diff
@@ -27,39 +33,188 @@
+
+Superjson solves these issues by providing a thin wrapper over `JSON.stringify` and `JSON.parse`. Look at the difference between manually converting invalid fields, and using Superjson to handle this for you:
+
+
+```js
+// 😔 without superjson
+
+// retrieve user object
+const user = await db.user.findOne();
+
+/* 
+user = {
+  id: 1,
+  createdAt: new Date(1),
+  updatedAt: new Date(2),
+  name: undefined,
+  email: 'me@here.com',
+  posts: [
+    {
+      id: 1,
+      createdAt: new Date(3),
+      updatedAt: new Date(4),
+      body: 'hello world!',
+    },
+  ],
+}; 
+*/
+
+// manually update invalid fields 
+const serializableUser = {
+  ...user,
+  createdAt: user.createdAt.toIsoString(),
+  updatedAt: user.updatedAt.toIsoString(),
+  name: 'undefined',
+  posts: user.posts.map(post => ({
+    ...post,
+    createdAt: post.createdAt.toIsoString(),
+    updatedAt: post.updatedAt.toIsoString(),
+  })),
+};
+
+// api response
+return JSON.stringify(serializableUser);
+
+// fetch user from api response
+const res = await fetch('/api/user');
+const json = await res.json();
+
+// restore fields
+const user = {
+  ...res,
+  createdAt: new Date(res.createdAt),
+  updatedAt: new Date(res.updatedAt),
+  name: undefined,
+  posts: res.posts.map(post => ({
+    ...post,
+    createdAt: new Date(post.createdAt),
+    updatedAt: new Date(post.createdAt),
+  })),
+};
+```
+
+```js
+// ✨ with superjson
+
+// retrieve user object
+const user = await db.user.findOne();
+
+/* 
+user = {
+  id: 1,
+  createdAt: new Date(1),
+  updatedAt: new Date(2),
+  name: undefined,
+  email: 'me@here.com',
+  posts: [
+    {
+      id: 1,
+      createdAt: new Date(3),
+      updatedAt: new Date(4),
+      body: 'hello world!',
+    },
+  ],
+}; 
+*/
+
+// api response
+return superjson.stringify(serializableUser);
+
+// fetch user from api response
+const res = await fetch('/api/user');
+
+// restore fields
+const user = superjson.parse(res);
+```
+
+## Getting started
+
+Install the library with your package manager of choice, e.g.:
+
+```
+yarn add superjson
+```
+
+## Usage
+
+The simplest way of using `superjson` is with its `stringify` and `parse` functions. If you know how to use `JSON.stringify` in JavaScript, you already know Superjson!
+
+Simply stringify any expression you’d like:
+
+```js
+import { stringify } from 'superjson';
+
+const json = stringify({date: new Date(0)});
+```
+
+And parse your JSON like so:
+
+```js
+import { parse } from 'superjson';
+
+const object = stringify(json);
+```
+
+Alternatively, you can use our lower-level `serializer` and `deserializer` functions. These transform any JavaScript expression into an object which is valid for JSON serialization.
```

Comment:
```suggestion
Alternatively, transform any JavaScript value into a JSON-compatible one by using our lower-level `serializer` and `deserializer` functions.
```

Reply:
(I like to put the desired outcome first before the solution, makes it easier to find what you're looking for)